bundles/org.openhab.binding.openuv/src/main/java/org/openhab/binding/openuv/internal/handler/OpenUVBridgeHandler.java

```diff
@@ -106,62 +94,50 @@ public void handleCommand(ChannelUID channelUID, Command command) {
         // Check if the provided api key is valid for use with the OpenUV service
-        getUVData("0", "0", null);
-    }
-
-    public Map<ThingUID, @Nullable ServiceRegistration<?>> getDiscoveryServiceRegs() {
-        return discoveryServiceRegs;
-    }
-
-    public void setDiscoveryServiceRegs(Map<ThingUID, @Nullable ServiceRegistration<?>> discoveryServiceRegs) {
-        this.discoveryServiceRegs = discoveryServiceRegs;
+        getUVData("0", "0", "0");
     }
 
-    @Override
-    public void handleRemoval() {
-        // removes the old registration service associated to the bridge, if existing
-        ServiceRegistration<?> dis = getDiscoveryServiceRegs().get(getThing().getUID());
-        if (dis != null) {
-            dis.unregister();
-        }
-        super.handleRemoval();
-    }
-
-    public @Nullable OpenUVResult getUVData(String latitude, String longitude, @Nullable String altitude) {
-        StringBuilder urlBuilder = new StringBuilder(BASE_URL).append("?lat=").append(latitude).append("&lng=")
-                .append(longitude);
-
-        if (altitude != null) {
-            urlBuilder.append("&alt=").append(altitude);
-        }
-        String errorMessage = null;
+    public @Nullable OpenUVResult getUVData(String latitude, String longitude, String altitude) {
         try {
-            String jsonData = HttpUtil.executeUrl("GET", urlBuilder.toString(), header, null, null, REQUEST_TIMEOUT);
+            String jsonData = HttpUtil.executeUrl("GET", String.format(QUERY_URL, latitude, longitude, altitude),
+                    header, null, null, REQUEST_TIMEOUT_MS);
             OpenUVResponse uvResponse = gson.fromJson(jsonData, OpenUVResponse.class);
             if (uvResponse.getError() == null) {
                 updateStatus(ThingStatus.ONLINE);
                 return uvResponse.getResult();
             } else {
-                errorMessage = uvResponse.getError();
+                throw new OpenUVException(uvResponse.getError());
             }
         } catch (IOException e) {
-            errorMessage = e.getMessage();
+            logger.error("Error occured during API query : {}", e.getMessage());
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        } catch (OpenUVException e) {
+            if (e.isQuotaError()) {
+                LocalDate today = LocalDate.now();
+                LocalDate tomorrow = today.plusDays(1);
+                LocalDateTime tomorrowMidnight = tomorrow.atStartOfDay().plusMinutes(2);
+
+                String message = "Quota Exceeded, going OFFLINE for today, will retry at : "
+                        + tomorrowMidnight.toString();
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, message);
+
+                scheduler.schedule(this::initiateConnexion,
+                        Duration.between(LocalDateTime.now(), tomorrowMidnight).toMinutes(), TimeUnit.MINUTES);
```

Comment:
You should keep track of this scheduled future so that you can cancel it if the handler gets disposed.